File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int cnt_ones=0,cnt_zeros=0;
        int anc=0,res=0;;
        for(int i=0;i<nums.size();i++)
        {
            if(nums[i]==1) cnt_ones++;
            else cnt_zeros++;
            while(anc<nums.size() && cnt_zeros>k){
                if(nums[anc]) cnt_ones--;
                else cnt_zeros--;
                anc++;
            }
            res=max(res,i-anc+1);
        }
        return res;
    }
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        zeros,ans,anc =0,0,0 
        for i,val in enumerate(nums): 
            if val==0: 
                zeros+=1
            while anc<=i and zeros>k: 
                if nums[anc]==0: 
                    zeros-=1 
                anc+=1 
            ans = max(ans,i-anc+1)
        return ans